### PR TITLE
feat: Star-search-tools releases aware

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

Makes the star-search tools able to query for releases. By default, it's releases over the last 30 days.

## Related Tickets & Documents

N/a - ongoing star-search work.

## Steps to QA

```sh
#!/bin/bash

url="http://localhost:3003/v2/star-search/stream"

curl -sN -X 'POST' "$url" \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "query_text": "What are the recent releases of kubernetes/kubernetes?"
}' | awk -F 'data: ' '/^data: / { printf "%s", $2 }'
```

```
  content: 'Here are the recent releases for the **kubernetes/kubernetes** repository:\n' +
    '\n' +
    '1. **Kubernetes v1.30.0**\n' +
    '   - Release Date: April 17, 2024\n' +
    '   - Tag: v1.30.0\n' +
    '   - Not a draft or pre-release.\n' +
    '\n' +
    '2. **Kubernetes v1.29.4**\n' +
    '   - Release Date: April 16, 2024\n' +
    '   - Tag: v1.29.4\n' +
    '   - Not a draft or pre-release.\n' +
    '\n' +
    '3. **Kubernetes v1.28.9**\n' +
    '   - Release Date: April 16, 2024\n' +
    '   - Tag: v1.28.9\n' +
    '   - Not a draft or pre-release.\n' +
    '\n' +
    '4. **Kubernetes v1.27.13**\n' +
    '   - Release Date: April 16, 2024\n' +
    '   - Tag: v1.27.13\n' +
    '   - Not a draft or pre-release.\n' +
    '\n' +
    '5. **Kubernetes v1.30.0-rc.2**\n' +
    '   - Release Date: April 11, 2024\n' +
    '   - Tag: v1.30.0-rc.2\n' +
    '   - This is marked as a pre-release.\n' +
    '\n' +
    'These releases were made from the `master` target reference, and all releases were handled by the `k8s-release-robot`.'
}
```

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: src/timescale/release_github_events.service.ts [hunk 1]
```diff
@@ -31,14 +31,28 @@ export class ReleaseGithubEventsService {
...
+  releasesQueryBuilder({
+    contributor = "",
+    notContributor = "",
+    repos = "",
+    repoIds = "",
+    range = 30,
+    order = OrderDirectionEnum.DESC,
+    prevDaysStartDate = 0,
+  }: {
+    contributor?: string;
```
Comment (nickytonline) [inline src/timescale/release_github_events.service.ts:43]: Just double checking, but is it expected that all parameters except `prevDaysStartDate` are optional?

**Review by nickytonline — APPROVED** [review]
Left one questiion, but LGTM.

:shipit:

**Review by bdougie — APPROVED** [review]
Exciting addition

**Review by brandonroberts — APPROVED** [review]
🚀

File: src/timescale/release_github_events.service.ts (hunk 1 above)
Comment (jpmcb) [reply src/timescale/release_github_events.service.ts:43]: Great question: this was a refactor so I wasn't repeating abunch of code in the release service. But, essentially, these are optional params, but _at least_ one of them is required. Which is why this check is here:

https://github.com/open-sauced/api/blob/b1a799112e412d9485bad914dc720dc82b179454/src/timescale/release_github_events.service.ts#L35-L37

**Pull request merged**